content/tutorials/reactjs-realtime-commenting.textile

```diff
@@ -0,0 +1,344 @@
+
+As you can publish a message to any channel, channels provide a means for you to filter data by topic or enforce access control for your subscribers. Only subscribers for those channesl will receive those messages.
+
+In this tutorial, I will be showing you how to use Ably "Realtime client library":/realtime to build a realtime commenting system with React. This tutorial will not include any server-side implementation. Hence, comments won't be persisted to any storage.
+
+Let's get started.
+
+<%= partial 'tutorials/_step-1-setup-free-account' %>
+
+h2. Step 2 - Create a React App
+
+We'll start by creating a new React app. To do this, we'll be using create-react-app. So you need to install @create-react-app@ locally if you don't have it already:
+
+```[sh]
+npm install -g create-react-app
+```
+
+Once installed, we can use it to create a new React app. We'll call it @reactjs-realtime-commenting@:
+
+```[sh]
+create-react-app reactjs-realtime-commenting
+```
+
+The new React app is ready. We can start it by running:
+
+```[sh]
+cd reactjs-realtime-commenting
+npm start
+```
+
+h2. Step 3 - Delete Unused Files
+
+Before we move on to building upon this app, let’s quickly delete some files we won’t be needing. So delete the following files: @src/App.css@, @src/App.test.js@, @src/index.css@ and @src/logo.svg@.
+
+h2. Step 4 - Create Component Folder
+
+Now, we’ll create a new folder within the @src@  folder called @components@. This folder will contain all our React components.
+
+```[sh]
+cd src
+mkdir components
+```
+
+Once that is created, we need to move @App.js@ into the @components@ folder:
+
+```[sh]
+cd src
+mv App.js components
+```
+
+h2. Step 5 - Update index.js
+
+Then we need to update @src/index.js@ as below:
+
+```[javascript]
+import React from 'react';
+import ReactDOM from 'react-dom';
+import App from './components/App';
+import registerServiceWorker from './registerServiceWorker';
+
+ReactDOM.render(<App />, document.getElementById('root'));
+registerServiceWorker();
+```
+
+h2. Step 6 - Add Bulma CSS Framework
+
+We’ll be using "Bulma":https://bulma.io Bulma CSS framework so as not to waste time writing CSS. So we’ll reference it from CDN. Add the line below to @public/index.html@:
+
+```[html]
+<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/bulma/0.6.0/css/bulma.min.css">
+```
+
+h2. Step 7 - Install Ably
+
+To start using Ably you first need to install the NPM module. The NPM module can be installed as follows:
+
+```[sh]
+npm install ably
+```
+
+The client library must be instanced with the API key you copied in Step 1. We’ll create a dedicated file for this so we can easily use it in multiple places. Within @src@ folder, create a new file named @ably.js@ and add the code below into it:
```

Comment:
Since the app is set up to use WebPack, it makes to use it this way to everything can be bundled together. This will reduce the number of HTTP requests made to fetch the library individually.

Reply:
OK